### PR TITLE
Rustier bindings

### PR DESCRIPTION
**Pull request by johncf**
I'm not very familiar with the codebase myself, and lack experience with gtk too. So I tried to follow the patterns found in the sources of `gtk-rs` crates.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -5,120 +5,153 @@
...
-// pub fn pango_cairo_font_map_create_context(fontmap: *mut PangoCairoFontMap) -> *mut pango::PangoContext;
-// pub fn pango_cairo_font_map_get_font_type(fontmap: *mut PangoCairoFontMap) -> cairo::enums::FontType;
-// pub fn pango_cairo_font_map_get_resolution(fontmap: *mut PangoCairoFontMap) -> c_double;
-// pub fn pango_cairo_font_map_set_default(fontmap: *mut PangoCairoFontMap);
-// pub fn pango_cairo_font_map_set_resolution(fontmap: *mut PangoCairoFontMap, dpi: c_double);
-
+impl PangoFontMapExt for pango::FontMap {
+    fn new() -> pango::FontMap {
+        unsafe { from_glib_full(ffi::pango_cairo_font_map_new()) }
+    }
```
Comment (johncf) [inline src/lib.rs]: These methods might look like they should go into [`font_map::FontMap`](https://github.com/johncf/pangocairo-rs/blob/ed0330e3/src/font_map.rs#L14) impl, but the API is confusing, including the docs. For instance, quoted from [`pango_cairo_font_map_new` docs](https://developer.gnome.org/pango/stable/pango-Cairo-Rendering.html#pango-cairo-font-map-new):

> Creates a new _PangoCairoFontMap_ object; a fontmap is used to cache information about available fonts, and holds certain global parameters such as the resolution.
>
> [...]
>
> ### Returns
>
> the newly allocated _PangoFontMap_, which should be freed with g_object_unref().

So ... the returned PangoFontMap can be used as a PangoCairoFontMap ???

Comment (RazrFalcon) [inline src/lib.rs]: Have no idea. As long as we can build it - it's fine. 

Some tests will be fine, probably.

**Comment by RazrFalcon** [comment]
Looks good. It definitely better than the previous version.

I'm not sure for with object we should implement traits: for cairo or for pango. But I think we should keep it like this for now.

File: src/lib.rs (hunk 1 above)
Comment (johncf) [inline src/lib.rs]: @EPashkin Would you mind sharing your thoughts on [this](https://github.com/RazrFalcon/pangocairo-rs/pull/3#discussion_r111682057)? And if interested, please comment on the [current state](https://github.com/RazrFalcon/pangocairo-rs/blob/master/src/lib.rs) of this crate. Any reason why pangocairo bindings are not "officially" supported? ([related issue](https://github.com/gtk-rs/pango/issues/31))

Comment (RazrFalcon) [inline src/lib.rs]: There are a lot of discussions here: https://github.com/gtk-rs/gtk/issues/453

Comment (johncf) [inline src/lib.rs]: @RazrFalcon Thanks!

Comment (EPashkin) [inline src/lib.rs]: IMHO PangoCairoFontMap wrapper need parent as it have `<prerequisite name="Pango.FontMap"/>`
`pub struct FontMap(Object<ffi::PangoCairoFontMap>): pango::FontMap;`
and this implementation go to it.
But as `pango::FontMap` currently have [implemented](https://github.com/gtk-rs/pango/blob/master/src/auto/font_map.rs#L16) with assumption 'no children for this class' it won't works currently and better this stay as is until pango fixed (this need add option to Gir beforehand).

**Pull request merged**